libs/wingc/src/type_check.rs

```diff
@@ -2622,40 +2649,78 @@ impl<'a> TypeChecker<'a> {
-							format!(
-								"bring {} must be assigned to an identifier (e.g. bring \"foo\" as foo)",
-								module_name.name
-							),
-						);
-						return;
+				match &source {
+					BringSource::BuiltinModule(name) => {
+						if WINGSDK_BRINGABLE_MODULES.contains(&name.name.as_str()) {
+							library_name = WINGSDK_ASSEMBLY_NAME.to_string();
+							namespace_filter = vec![name.name.clone()];
+							alias = identifier.as_ref().unwrap_or(&name);
+						} else if name.name.as_str() == WINGSDK_STD_MODULE {
+							self.spanned_error(stmt, format!("Redundant bring of \"{}\"", WINGSDK_STD_MODULE));
+							return;
+						} else {
+							self.spanned_error(stmt, format!("\"{}\" is not a built-in module", name));
+							return;
+						}
 					}
-					// We assume we have a jsii library and we use `module_name` as the library name, and set no
-					// namespace filter (we only support importing a full library at the moment)
-					library_name = module_name.name[1..module_name.name.len() - 1].to_string();
-					namespace_filter = vec![];
-					alias = identifier.as_ref().unwrap();
-				} else {
-					// case 2: bring module_name;
-					// case 3: bring module_name as identifier;
-					if WINGSDK_BRINGABLE_MODULES.contains(&module_name.name.as_str()) {
-						library_name = WINGSDK_ASSEMBLY_NAME.to_string();
-						namespace_filter = vec![module_name.name.clone()];
-						alias = identifier.as_ref().unwrap_or(&module_name);
-					} else if module_name.name.as_str() == WINGSDK_STD_MODULE {
-						self.spanned_error(stmt, format!("Redundant bring of \"{}\"", WINGSDK_STD_MODULE));
-						return;
-					} else {
-						self.spanned_error(stmt, format!("\"{}\" is not a built-in module", module_name.name));
+					BringSource::JsiiModule(name) => {
+						if identifier.is_none() {
+							self.spanned_error(
+								stmt,
+								format!(
+									"bring \"{}\" must be assigned to an identifier (e.g. bring \"foo\" as foo)",
+									name
+								),
+							);
+							return;
+						}
+						library_name = name.name.to_string();
+						// no namespace filter (we only support importing entire libraries at the moment)
+						namespace_filter = vec![];
+						alias = identifier.as_ref().unwrap();
+					}
+					BringSource::WingFile(name) => {
+						if identifier.is_none() {
+							self.spanned_error(
+								stmt,
+								format!(
+									"bring \"{}\" must be assigned to an identifier (e.g. bring \"foo\" as foo)",
+									name
+								),
+							);
+							return;
+						}
+
+						let lookup_name = sanitize_dots(&name.name);
+						let lookup_result = self.types.libraries.lookup_nested_str(&lookup_name, None);
+						match lookup_result {
+							LookupResult::Found(kind, _) => match kind {
+								SymbolKind::Type(_) => panic!("expected a namespace, found a type"),
+								SymbolKind::Variable(_) => panic!("expected a namespace, found a variable"),
+								SymbolKind::Namespace(nsref) => {
+									if !nsref.is_bringable {
```

Comment:
Could be cool is instead of a `bool` here we'd have some span info of the invalid statement, and generate an appropriate diag.

Reply:
Created an issue https://github.com/winglang/wing/issues/3626